OloEngine/src/OloEngine/Asset/AssetSerializer.cpp

```
@@ -59,6 +61,122 @@ namespace OloEngine
+        // Create the texture on the main thread (GL calls happen here)
+        Ref<Texture2D> texture = Texture2D::Create(spec);
+
+        if (!texture)
+        {
+            OLO_CORE_ERROR("TextureSerializer::FinalizeFromRawData - Failed to create texture: {}", texData.DebugName);
+            return false;
+        }
+
+        // Set the pixel data on the texture
+        // Note: SetData expects size in bytes
+        u32 dataSize = static_cast<u32>(texData.PixelData.size());
```

Clang (14.0.6)

[warning] variable 'dataSize' is not initialized

(cppcoreguidelines-init-variables)